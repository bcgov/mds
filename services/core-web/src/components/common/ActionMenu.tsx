import { Button, Dropdown } from "antd";
import { CARAT } from "@/constants/assets";
import React, { FC } from "react";
import { ITableAction } from "@/components/common/CoreTableCommonColumns";

interface ActionMenuProps {
  record: any;
  actionItems: ITableAction[];
  category: string;
}
const ActionMenu: FC<ActionMenuProps> = ({ record, actionItems, category }) => {
  const items = actionItems.map((action) => {
    return {
      key: action.key,
      icon: action.icon,
      label: (
        <button
          type="button"
          className={`full actions-dropdown-button`}
          onClick={(event) => action.clickFunction(event, record)}
        >
          <div>{action.label}</div>
        </button>
      ),
    };
  });
  return (
    <Dropdown menu={{ items }} placement="bottomLeft">
      <Button
        icon={
          <img
            className="padding-sm--right icon-svg-filter"
            src={CARAT}
            alt={`${category} Actions`}
            style={{ paddingLeft: "5px" }}
          />
        }
        type="text"
        className="permit-table-button"
      >
        Actions
      </Button>
    </Dropdown>
  );
};

export default ActionMenu;
