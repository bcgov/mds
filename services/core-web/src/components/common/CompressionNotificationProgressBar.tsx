import React, { FC } from "react";
import { Progress } from "antd";

interface CompressionNotificationProgressBarProps {
  compressionProgress: number;
  notificationTopPosition: number;
}

export const CompressionNotificationProgressBar: FC<CompressionNotificationProgressBarProps> = (
  props
) => (
  <Progress
    percent={props.compressionProgress}
    showInfo={false}
    strokeColor={"#5e46a1"}
    strokeLinecap={"square"}
    trailColor="#d9d9d9"
    className="compressionProgressBar"
    style={{
      top: `${props.notificationTopPosition - 10}px`,
    }}
  />
);

export default CompressionNotificationProgressBar;
