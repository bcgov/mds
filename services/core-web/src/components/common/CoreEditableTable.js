import React from "react";
import { connect } from "react-redux";
import { PropTypes } from "prop-types";
import { getNoticeOfWorkUnitTypeOptionsHash } from "@common/selectors/staticContentSelectors";
import { Field, FieldArray } from "redux-form";
import { Button } from "antd";
import * as Strings from "@common/constants/strings";
import { TRASHCAN } from "@/constants/assets";
import "@ant-design/compatible/assets/index.css";

const propTypes = {
  isViewMode: PropTypes.bool.isRequired,
  fieldName: PropTypes.string.isRequired,
  tableContent: PropTypes.objectOf(PropTypes.any).isRequired,
  fieldID: PropTypes.string.isRequired,
  type: PropTypes.string,
  unitTypeHash: PropTypes.objectOf(PropTypes.string).isRequired,
};

const defaultProps = {
  type: "Activity",
};

const addActivity = (fields) => {
  fields.push({});
};

const removeActivity = (fields, index, fieldID) => {
  if (fields.get(index) && fields.get(index)[fieldID]) {
    // add state_modified and set to "delete" for backend
    fields.get(index).state_modified = "delete";

    // move updated object, this will cause rerendering of the react component, setTimeout is required to bypass react optimization
    setTimeout(() => {
      // eslint-disable-next-line no-constant-condition
      const res = fields.move(index, (index = 0 ? index + 1 : index - 1));
      return res;
    }, 1);
  } else {
    fields.remove(index);
  }
};

const renderActivities = ({ fields, isViewMode, tableContent, type, fieldID, unitTypeHash }) => {
  // resets deleted state if users decided to cancel their changes
  if (isViewMode && fields.length !== 0) {
    fields.getAll().forEach((activity) => {
      // eslint-disable-next-line no-prototype-builtins
      if (activity && activity.hasOwnProperty("state_modified")) {
        delete activity.state_modified;
      }
    });
  }

  const activeRecordsCount =
    fields.length !== 0 ? fields.getAll().filter((activity) => !activity.state_modified).length : 0;

  const renderViewField = (fieldObj, index, content) => {
    const activityObj = fieldObj.get(index);
    if (content.hasUnit) {
      const unit = activityObj[`${content.value}_unit_type_code`]
        ? unitTypeHash[activityObj[`${content.value}_unit_type_code`]]
        : Strings.EMPTY_FIELD;
      return `${activityObj[content.value]} ${unit}`;
    }
    return activityObj[content.value];
  };

  return (
    <div>
      <div className="ant-table-wrapper" style={{ position: "relative", zIndex: 5 }}>
        <div className={`ant-table ${(!fields || fields.length <= 0) && "ant-table-empty"}`}>
          <div className="ant-table-content">
            <table style={{ tableLayout: "auto" }}>
              <thead className="ant-table-thead">
                <tr>
                  {tableContent.map(
                    ({ title, isUnit }) =>
                      (!isUnit || !isViewMode) && <th className="ant-table-cell">{title}</th>
                  )}
                  <th className="ant-table-cell" />
                </tr>
              </thead>
              <tbody className="ant-table-tbody">
                {activeRecordsCount > 0 &&
                  fields.map((activity, index) => {
                    const activityObj = fields.get(index);
                    const key = activityObj && (activityObj[fieldID] || index);
                    return (
                      (isViewMode || (activityObj && !activityObj.state_modified)) && (
                        <tr className="ant-table-row ant-table-row-level-0" key={key}>
                          {tableContent.map((content) => {
                            return isViewMode ? (
                              !content.isUnit && (
                                <td className="ant-table-cell">
                                  <div title={content.title}>
                                    {renderViewField(fields, index, content, tableContent)}
                                  </div>
                                </td>
                              )
                            ) : (
                              <td className="ant-table-cell">
                                <div title={content.title}>
                                  <Field
                                    name={`${activity}.${content.value}`}
                                    value={`${activity}.${content.value}`}
                                    component={content.component}
                                    disabled={isViewMode}
                                    validate={content.validate}
                                    data={content.data || []}
                                    minRows={content.minRows}
                                  />
                                </div>
                              </td>
                            );
                          })}
                          {!isViewMode && (
                            <td className="ant-table-cell">
                              <div name="remove" title="remove">
                                <Button
                                  type="primary"
                                  size="small"
                                  onClick={() => removeActivity(fields, index, fieldID)}
                                  ghost
                                >
                                  <img name="remove" src={TRASHCAN} alt="Remove Activity" />
                                </Button>
                              </div>
                            </td>
                          )}
                          {isViewMode && <td className="ant-table-cell" />}
                        </tr>
                      )
                    );
                  })}
                {activeRecordsCount <= 0 && (
                  <tr className="ant-table-placeholder">
                    <td colSpan="100%" className="ant-table-cell" style={{ color: "#bfbfbf" }}>
                      No Data Yet
                    </td>
                  </tr>
                )}
              </tbody>
            </table>
          </div>
        </div>
      </div>
      {!isViewMode && (
        <div>
          <Button type="primary" onClick={() => addActivity(fields)}>
            {`Add ${type}`}
          </Button>
        </div>
      )}
    </div>
  );
};

export const CoreEditableTable = (props) => {
  return (
    <div>
      <FieldArray
        name={props.fieldName}
        component={renderActivities}
        rerenderOnEveryChange
        {...{
          isViewMode: props.isViewMode,
          tableContent: props.tableContent,
          type: props.type,
          fieldID: props.fieldID,
          unitTypeHash: props.unitTypeHash,
        }}
      />
    </div>
  );
};

CoreEditableTable.propTypes = propTypes;
CoreEditableTable.defaultProps = defaultProps;

export default connect(
  (state) => ({
    unitTypeHash: getNoticeOfWorkUnitTypeOptionsHash(state),
  }),
  null
)(CoreEditableTable);
