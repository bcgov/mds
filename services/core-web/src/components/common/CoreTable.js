import React from "react";
import { PropTypes } from "prop-types";
import { Table } from "antd";
import NullScreen from "@/components/common/NullScreen";

/**
 * @constant CoreTable renders react children or a skeleton loading view using the column headers
 * of the tables and 9 loading rows
 *
 * condition = expecting a truthy value to render children, ie "isLoaded"
 * tableHeaders = array of table headers, the skeleton loading will have the same number of columns and same titles
 * isPaginated = true if the table is paginated - this allows the skeleton to be smooth when changing pages
 *
 *
 */

const propTypes = {
  columns: PropTypes.arrayOf(PropTypes.object).isRequired,
  dataSource: PropTypes.objectOf(PropTypes.any).isRequired,
  condition: PropTypes.bool.isRequired,
  tableProps: PropTypes.objectOf(PropTypes.any),
};

const defaultProps = {
  tableProps: {},
};

export const CoreTable = (props) => {
  const baseProps = {
    locale: { emptyText: <NullScreen type="no-results" /> },
    pagination: false,
    rowClassName: "fade-in",
    tableLayout: "auto",
  };
  const combinedProps = { ...baseProps, ...props.tableProps };
  const skeletonColumns = props.columns.map((column) => ({
    title: column.title,
    dataIndex: column.title,
    className: column.className,
    width: column.width,
    render: () => <div className={`skeleton-table__loader ${column.className}`} />,
  }));

  return (
    <div>
      {props.condition ? (
        <div>
          <Table {...combinedProps} columns={props.columns} dataSource={props.dataSource} />
        </div>
      ) : (
        <div className="skeleton-table">
          <Table
            align="left"
            pagination={false}
            columns={skeletonColumns}
            dataSource={new Array(props.tableProps.pagination ? 25 : 9).fill({})}
            rowClassName="skeleton-table__row"
          />
        </div>
      )}
    </div>
  );
};

CoreTable.propTypes = propTypes;
CoreTable.defaultProps = defaultProps;

export default CoreTable;
