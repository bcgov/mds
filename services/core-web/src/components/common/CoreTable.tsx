import React from "react";
import { Table, TableProps, Tooltip } from "antd";
import { ColumnsType } from "antd/es/table";
import { MinusSquareFilled, PlusSquareFilled } from "@ant-design/icons";
import { ExpandableConfig, TableRowSelection } from "antd/lib/table/interface";

interface CoreTableExpandConfig<T> extends ExpandableConfig<T> {
  getDataSource?: (record: T) => any[];
  rowKey?: string | ((record: any) => string);
  recordDescription?: string;
  subTableColumns?: ColumnsType<any>;
  showVersionHistory?: boolean;
  // and any other props from expandable https://4x.ant.design/components/table/#expandable
}

interface CoreTableProps<T> extends TableProps<T> {
  columns: ColumnsType<T>;
  dataSource: T[];
  condition?: boolean;
  rowKey?: string | ((record: T) => string | number); // defaults to "key"
  classPrefix?: string;
  emptyText?: string;
  expandProps?: CoreTableExpandConfig<any> | null;
  rowSelection?: TableRowSelection<any>;
}

const CoreTable = <T,>(props: CoreTableProps<T>) => {
  const {
    columns,
    condition = true,
    pagination = false,
    tableLayout = "auto",
    emptyText = "No Data Yet",
    expandProps = null,
    classPrefix = "",
    ...tableProps
  } = props;

  const skeletonColumns = columns.map((column) => ({
    title: column.title,
    className: column.className,
    width: column.width,
    render: () => <div className={`skeleton-table__loader ${column.className}`} />,
  }));

  const tableClass = classPrefix ? classPrefix + "-table" : "";

  const renderTableExpandIcon = ({ expanded, onExpand, record }) => {
    const { rowExpandable, getDataSource, recordDescription = "details" } = expandProps;
    const isRowExpandable = rowExpandable ?? ((rec) => getDataSource(rec).length > 0);
    if (!isRowExpandable(record)) {
      return null;
    }
    return (
      <a
        role="link"
        className="expand-row-icon"
        onClick={(e) => onExpand(record, e)}
        style={{ cursor: "pointer" }}
        tabIndex={0}
      >
        {expanded ? (
          <Tooltip
            title={`Click to hide ${recordDescription}.`}
            placement="right"
            mouseEnterDelay={1}
          >
            <MinusSquareFilled className="icon-lg--lightgrey" />
          </Tooltip>
        ) : (
          <Tooltip
            title={`Click to show ${recordDescription}.`}
            placement="right"
            mouseEnterDelay={1}
          >
            <PlusSquareFilled className="icon-lg--lightgrey" />
          </Tooltip>
        )}
      </a>
    );
  };

  const renderExpandedRow = (record: any) => {
    return (
      <Table
        columns={expandProps.subTableColumns}
        dataSource={expandProps.getDataSource(record)}
        locale={{ emptyText }}
        pagination={false}
        size="small"
        className={`${tableClass} nested-table`}
        rowClassName={`${tableClass} expanded-row fade-in`}
        rowKey={expandProps.rowKey}
      />
    );
  };

  const getExpansionProps = () => {
    if (expandProps) {
      return expandProps.showVersionHistory
        ? { expandIcon: renderTableExpandIcon, indentSize: 0, ...expandProps }
        : {
            rowExpandable:
              expandProps.rowExpandable ??
              ((record) => expandProps.getDataSource(record).length > 0),
            expandIcon: renderTableExpandIcon,
            expandRowByClick: true,
            expandedRowRender: expandProps.expandedRowRender ?? renderExpandedRow,
            ...expandProps,
          };
    }
    return { showExpandColumn: false };
  };

  return condition ? (
    <Table
      expandable={getExpansionProps()}
      pagination={pagination}
      locale={{ emptyText }}
      className={`${tableClass} core-table`}
      tableLayout={tableLayout}
      rowClassName={
        expandProps?.showVersionHistory
          ? "table-row-align-middle no-sub-table-expandable-rows fade-in"
          : "fade-in"
      }
      {...(expandProps?.showVersionHistory && tableProps?.rowSelection
        ? {
            rowSelection: { ...tableProps.rowSelection },
          }
        : {})}
      columns={columns}
      {...tableProps}
    ></Table>
  ) : (
    <Table
      pagination={false}
      columns={skeletonColumns}
      dataSource={Array(pagination ? 25 : 9)}
      rowClassName="skeleton-table__row"
    ></Table>
  );
};

export default CoreTable;
