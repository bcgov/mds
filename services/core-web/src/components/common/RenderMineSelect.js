import React, { Component } from "react";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { AutoComplete } from "antd";
import { connect } from "react-redux";
import { fetchMineNameList } from "@common/actionCreators/mineActionCreator";
import { getMineWithoutStore } from "@common/utils/actionlessNetworkCalls";
import { getMineNames } from "@common/selectors/mineSelectors";
import RenderAutoComplete from "@/components/common/RenderAutoComplete";
import MineCard from "@/components/mine/NoticeOfWork/MineCard";
import CustomPropTypes from "@/customPropTypes";
import LoadingWrapper from "@/components/common/wrappers/LoadingWrapper";
/**
 * @constant RenderMineSelect - Ant Design `AutoComplete` component for redux-form that handles all the state for selecting a mine and updating a content card with that mines tombstone data.
 *
 */

const propTypes = {
  // redux-form value id
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  input: PropTypes.objectOf(PropTypes.any),
  meta: PropTypes.objectOf(PropTypes.any),
  label: PropTypes.String,
  fetchMineNameList: PropTypes.func.isRequired,
  mineNameList: PropTypes.arrayOf(CustomPropTypes.mineName).isRequired,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  showCard: PropTypes.bool,
  majorMineOnly: PropTypes.bool,
  onMineSelect: PropTypes.func,
  fullWidth: PropTypes.bool,
  additionalPin: PropTypes.arrayOf(PropTypes.string),
};

const defaultProps = {
  placeholder: "Search for a mine by name",
  disabled: false,
  showCard: false,
  majorMineOnly: undefined,
  onMineSelect: () => {},
  meta: {},
  label: "Select a mine",
  input: { value: "" },
  fullWidth: false,
  additionalPin: [],
};

export class RenderMineSelect extends Component {
  state = {
    selectedMine: false,
  };

  componentDidMount() {
    if (this.props.input.value) {
      getMineWithoutStore(this.props.input.value).then((data) => {
        this.setState({ selectedMine: data.data });
        this.handleChange(data.data.mine_name);
      });
    }
  }

  handleChange = (name) => {
    let params = { name };
    if (this.props.majorMineOnly) {
      params = { major: true, ...params };
    }

    if (name.length > 2) {
      this.props.fetchMineNameList(params);
    } else if (name.length === 0) {
      this.props.fetchMineNameList({ major: true });
    }
  };

  handleSelect = (value) => {
    this.setState({ selectedMine: false });
    getMineWithoutStore(value).then((data) => {
      this.setState({ selectedMine: data.data });
      this.props.onMineSelect(value);
    });
  };

  transformData = (data) =>
    data.map(({ mine_guid, mine_name, mine_no }) => (
      <AutoComplete.Option key={mine_guid} value={mine_guid}>
        {`${mine_name} - ${mine_no}`}
      </AutoComplete.Option>
    ));

  render() {
    const isLoaded = this.state.selectedMine || false;
    const width = this.props.fullWidth ? "93vw" : "100%";
    return (
      <div>
        <RenderAutoComplete
          {...this.props}
          placeholder="Search for a mine by name"
          handleSelect={this.handleSelect}
          data={this.transformData(this.props.mineNameList)}
          handleChange={this.handleChange}
        />
        <div style={{ position: "relative", height: "inherit", width }}>
          <LoadingWrapper condition={isLoaded}>
            <MineCard mine={this.state.selectedMine} additionalPin={this.props.additionalPin} />
          </LoadingWrapper>
        </div>
      </div>
    );
  }
}
const mapStateToProps = (state) => ({
  mineNameList: getMineNames(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchMineNameList,
    },
    dispatch
  );

RenderMineSelect.propTypes = propTypes;
RenderMineSelect.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(RenderMineSelect);
