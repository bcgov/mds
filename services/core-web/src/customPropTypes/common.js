import { PropTypes, shape, arrayOf, objectOf, oneOfType } from "prop-types";

export const dropdownListItem = shape({
  value: oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  label: PropTypes.string.isRequired,
});

export const filterListItem = shape({
  value: oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  text: PropTypes.string.isRequired,
});

export const groupedDropdownList = shape({
  groupName: oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  opt: arrayOf(dropdownListItem),
});

export const options = arrayOf(dropdownListItem);
export const groupOptions = arrayOf(groupedDropdownList);
export const filterOptions = arrayOf(filterListItem);

export const formMeta = shape({
  touched: PropTypes.bool,
  error: PropTypes.string,
  warning: PropTypes.string,
});

export const match = shape({
  isExact: PropTypes.bool,
  params: objectOf(PropTypes.string),
  path: PropTypes.string,
  url: PropTypes.string,
});

export const genericFormState = shape({
  anyTouched: PropTypes.bool,
  registeredFields: objectOf(objectOf(oneOfType([PropTypes.string, PropTypes.number]))),
  fields: objectOf(objectOf(PropTypes.bool)),
  syncErrors: PropTypes.string,
  values: objectOf(PropTypes.string),
});

export const pageData = shape({
  current_page: PropTypes.number,
  items_per_page: PropTypes.number,
  records: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  total: PropTypes.number,
  total_pages: PropTypes.number,
});
