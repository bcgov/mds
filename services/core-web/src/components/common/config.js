import RenderCheckbox from "./RenderCheckbox";
import RenderGroupCheckbox from "./RenderGroupCheckbox";
import RenderAutocomplete from "./RenderAutoComplete";
import RenderAutoSizeField from "./RenderAutoSizeField";
import RenderCascader from "./RenderCascader";
import RenderDate from "./RenderDate";
import RenderTime from "./RenderTime";
import RenderYear from "./RenderYear";
import RenderField from "./RenderField";
import RenderScrollField from "./RenderScrollField";
import RenderLargeSelect from "./RenderLargeSelect";
import RenderSelect from "./RenderSelect";
import RenderMultiSelect from "./RenderMultiSelect";
import RenderRadioButtons from "./RenderRadioButtons";
import RenderGroupedSelect from "./RenderGroupedSelect";
import RenderMineSelect from "./RenderMineSelect";
import RenderLabel from "./RenderLabel";

// This file is anticipated to have multiple exports
// eslint-disable-next-line import/prefer-default-export
export const renderConfig = {
  CHECKBOX: RenderCheckbox,
  GROUP_CHECK_BOX: RenderGroupCheckbox,
  AUTOCOMPLETE: RenderAutocomplete,
  AUTO_SIZE_FIELD: RenderAutoSizeField,
  CASCADER: RenderCascader,
  DATE: RenderDate,
  TIME: RenderTime,
  YEAR: RenderYear,
  FIELD: RenderField,
  SCROLL_FIELD: RenderScrollField,
  SELECT: RenderSelect,
  LARGE_SELECT: RenderLargeSelect,
  MULTI_SELECT: RenderMultiSelect,
  RADIO: RenderRadioButtons,
  GROUPED_SELECT: RenderGroupedSelect,
  MINE_SELECT: RenderMineSelect,
  LABEL: RenderLabel,
};
