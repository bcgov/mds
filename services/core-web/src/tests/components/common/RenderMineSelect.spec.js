import React from "react";
import { shallow } from "enzyme";
import { RenderMineSelect } from "@/components/common/RenderMineSelect";
import { PARTY } from "@/tests/mocks/dataMocks";

let props = {};

const setupProps = () => {
  props = {
    id: "1",
    input: "",
    label: "",
    type: "",
    meta: {
      touched: false,
      error: false,
      warning: false,
    },
    data: [],
    option: {},
    name: "field",
    fetchMineNameList: jest.fn(),
    mineNameList: [],
    placeholder: "Select",
    disabled: false,
    majorMineOnly: false,
    onMineSelect: jest.fn(),
    fullWidth: true,
    additionalPin: [],
  };
};

beforeEach(() => {
  setupProps();
});

describe("RenderMineSelect", () => {
  it("renders properly", () => {
    props.data = PARTY.partyIds;
    props.option = PARTY.parties;
    const wrapper = shallow(<RenderMineSelect {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
