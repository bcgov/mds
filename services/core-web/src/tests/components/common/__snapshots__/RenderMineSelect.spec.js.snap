// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderMineSelect renders properly 1`] = `
<div>
  <RenderAutoComplete
    additionalPin={Array []}
    data={Array []}
    disabled={false}
    fetchMineNameList={[MockFunction]}
    fullWidth={true}
    handleChange={[Function]}
    handleSelect={[Function]}
    id="1"
    input=""
    label=""
    majorMineOnly={false}
    meta={
      Object {
        "error": false,
        "touched": false,
        "warning": false,
      }
    }
    mineNameList={Array []}
    name="field"
    onMineSelect={[MockFunction]}
    option={
      Object {
        "18133c75-49ad-4101-85f3-a43e35ae989a": Object {
          "address": Array [
            Object {},
          ],
          "business_role_appts": Array [
            Object {},
          ],
          "party_guid": "18133c75-49ad-4101-85f3-a43e35ae989a",
          "party_name": "mock name",
          "party_type_code": "PER",
        },
        "18145c75-49ad-0101-85f3-a43e45ae989a": Object {
          "address": Array [
            Object {},
          ],
          "business_role_appts": Array [
            Object {},
          ],
          "party_guid": "18145c75-49ad-0101-85f3-a43e45ae989a",
          "party_name": "mock Two",
          "party_type_code": "PER",
        },
      }
    }
    placeholder="Search for a mine by name"
    type=""
  />
  <div
    style={
      Object {
        "height": "inherit",
        "position": "relative",
        "width": "93vw",
      }
    }
  >
    <LoadingWrapper
      condition={false}
    >
      <Connect(MineCard)
        additionalPin={Array []}
        mine={false}
      />
    </LoadingWrapper>
  </div>
</div>
`;
